### PR TITLE
Chat opening hours

### PR DESCRIPTION
**Pull request by BreathingFlesh**
Please fill in the commit message below and work through the checklist. You can delete parts that are not needed, e.g. the optional description, the link to a ticket or irrelevant options of the checklist.

## Commit message

Town6: Adjustable opening hours for chat

TYPE: Feature

## Checklist

- [x] I have performed a self-review of my code
- [x] I considered adding a reviewer
- [x] I have updated the PO files
- [x] I made changes/features for both org and town6
- [x] I have updated the election_day API docs
- [x] I have tested my code thoroughly by hand
    - [x] I have tested styling changes/features on different browsers
    - [x] I have tested javascript changes/features on different browsers
- [x] I have added tests for my changes/features
    - [x] I am using freezegun for tests depending on date and times

### CODE REVIEW

File: src/onegov/town6/views/chat.py [hunk 1]
```diff
@@ -117,7 +117,9 @@ def view_chats_archive(self, request):
     form=ChatInitiationForm)
 def view_chat_form(self: ChatCollection, request, form):
 
-    if not request.app.chat_active and not request.is_manager:
+    if not request.app.chat_open(
+        request
+    ) and not request.is_manager:
```
Comment (cyrillkuettel) [inline src/onegov/town6/views/chat.py:122]: This fits on one line
```suggestion
if not request.app.chat_active and not request.is_manager:
```

File: src/onegov/town6/forms/settings.py [hunk 2]
```diff
@@ -143,5 +172,71 @@ def populate_chat_staff(self):
...
+                4: self.request.translate(_("Fr")),
+                5: self.request.translate(_("Sa")),
+                6: self.request.translate(_("Su"))
+            }
+        })
+
     def on_request(self):
         self.populate_chat_staff()
+        for include in self.on_request_include:
+            self.request.include(include)
```
Comment (cyrillkuettel) [inline src/onegov/town6/forms/settings.py:242]: We put this here in an experiment, I suppose it's not necessary now that we used the linking from org approach.
```suggestion
```

File: src/onegov/town6/forms/settings.py [hunk 3]
```diff
@@ -118,20 +121,46 @@ class ChatSettingsForm(Form):
...
+        super().__init__(*args, **kwargs)
+        self.time_errors = {}
+
     def process_obj(self, obj):
         super().process_obj(obj)
         self.chat_staff = obj.chat_staff or {}
         self.enable_chat = obj.enable_chat or {}
+        self.specific_opening_hours = obj.specific_opening_hours or {}
+        # self.opening_hours_chat.data = self.time_to_json(None)
+        if not obj.opening_hours_chat or None:
```
Comment (cyrillkuettel) [inline src/onegov/town6/forms/settings.py:150]: What's the point of `or None`? Shouldn't this just be 

```suggestion
        if not obj.opening_hours_chat:
```

File: src/onegov/town6/forms/settings.py [hunk 4]
```diff
@@ -118,20 +121,46 @@ class ChatSettingsForm(Form):
...
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.time_errors = {}
+
     def process_obj(self, obj):
         super().process_obj(obj)
         self.chat_staff = obj.chat_staff or {}
         self.enable_chat = obj.enable_chat or {}
+        self.specific_opening_hours = obj.specific_opening_hours or {}
+        # self.opening_hours_chat.data = self.time_to_json(None)
```
Comment (cyrillkuettel) [inline src/onegov/town6/forms/settings.py:149]: Can remove comment?
```suggestion
```

File: src/onegov/town6/forms/settings.py [hunk 5]
```diff
@@ -110,6 +112,7 @@ def clear_locale(response):
 
 
 class ChatSettingsForm(Form):
+    on_request_include = ()
```
Comment (cyrillkuettel) [inline src/onegov/town6/forms/settings.py:115]: We better get rid of this as well .
```suggestion
```

File: src/onegov/town6/forms/settings.py [hunk 6]
```diff
@@ -143,5 +172,71 @@ def populate_chat_staff(self):
             (v, k) for v, k in staff_members
         ]
 
+    def validate(self):
+        result = super().validate()
+        if self.specific_opening_hours.data:
```
Comment (cyrillkuettel) [inline src/onegov/town6/forms/settings.py:177]: Small nitpick, but you could save the result of `json_to_time` so we don't have to compute it twice.

File: src/onegov/town6/forms/chat.py [hunk 7]
```diff
@@ -41,8 +41,8 @@ class ChatInitiationForm(Form):
     confirmation = BooleanField(
         label=_("Confirmation"),
         description=_(
-            "Ich bestätige, dass mir bewusst ist, dass dieser Chat "
-            "gespeichert und der Verlauf an mich per Mail gesendet wird."),
+            "I confirm that I am aware that this chat will be saved and the "
```
Comment (cyrillkuettel) [inline src/onegov/town6/forms/chat.py:44]: Screenshot from 2024-01-29 11-19-57

translation is showing english.

**Review by cyrillkuettel — COMMENTED** [review]
Seems good overall, just a couple of nitpicks

File: src/onegov/town6/forms/settings.py (hunk 2 above)
Comment (BreathingFlesh) [reply src/onegov/town6/forms/settings.py:242]: Good point, I'll try it without.

File: src/onegov/town6/forms/settings.py (hunk 3 above)
Comment (BreathingFlesh) [reply src/onegov/town6/forms/settings.py:150]: True, this doesn't do anything.

**Pull request merged**